src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -924,4 +933,15 @@ class LicenceService(
     val parts = version.split(".")
     return Pair(parts[0].toInt(), parts[1].toInt())
   }
+
+  private fun inactivateInProgressLicenceVersions(licences: List<EntityLicence>) {
+    val licenceIds = licences.map { it.id }
+    val licencesToDeactivate =
+      licenceRepository.findAllByVersionOfIdInAndStatusCodeIn(licenceIds, listOf(IN_PROGRESS, SUBMITTED))
```

Comment:
is it worth capturing the reason why we're inactivating them or is it going to be reasonably apparent?

Reply:
yes, again I did have this before when I was calling this for each licence. I did an optimisation to process a list of all licences at once and removed the reason string for some reason. I'll have a look at adding it back

Reply:
worth doing as it will make support easier, so added a reason param